server/app/controllers/admin/RequestChecker.java

```diff
@@ -0,0 +1,25 @@
+package controllers.admin;
+
+import javax.inject.Inject;
+import repository.VersionRepository;
+
+/** Utility class for checking Request parameters. */
+class RequestChecker {
+  private VersionRepository versionRepository;
+
+  @Inject
+  public RequestChecker(VersionRepository versionRepository) {
+    this.versionRepository = versionRepository;
+  }
+
+  /**
+   * Verifies programId is associated with the Draft version and throws {@code
+   * NotChangeableException} if not.
+   */
+  void throwIfNotDraft(Long programId) {
+    if (versionRepository.isDraftProgram(programId)) {
```

Comment:
suggestion (blocking): since the method signature doesn't specify the resource type (just a `Long`) the method name should since there are other resources that can be in draft mode.
```suggestion
  void throwIfProgramNotDraft(Long programId) {
    if (versionRepository.isDraftProgram(programId)) {
```